crates/bevy_reflect/src/fields.rs

```diff
@@ -24,20 +28,32 @@ impl NamedField {
     pub fn type_name(&self) -> &str {
         self.type_name.borrow()
     }
+
+    /// The `TypeId` of the field
+    pub fn type_id(&self) -> TypeId {
+        self.type_id
+    }
```

Comment:
All the info structs contain this method. Should it be renamed to something else since it's shadowing the `type_id()` method from `std::any::Any`?

Other options include `ty()`, `id()`, and `inner_type()`. Or any other suggestion.

Personally I like `type_id()` because the user almost never wants the `TypeId` of the info struct— they want the underlying type. And it also makes naming much more consistent (e.g., `type_id()`, `item_type_id()`, `type_name()`). But I know some people may disagree with this.